src/Symfony/Component/VarDumper/Tests/Caster/MemcachedCasterTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Tests\Caster;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\VarDumper\Caster\MemcachedCaster;
+use Symfony\Component\VarDumper\Test\VarDumperTestTrait;
+
+class MemcachedCasterTest extends TestCase
+{
+    use VarDumperTestTrait;
+
+    public function testCastMemcachedWithDefaultOptions()
+    {
+        if (!class_exists('Memcached')) {
+            $this->markTestSkipped('Memcached not available');
+        }
+
+        $var = new \Memcached();
+        $var->addServer('127.0.0.1', 11211);
+        $var->addServer('127.0.0.2', 11212);
+
+        $expected = <<<EOTXT
+Memcached {
+  servers: array:2 [
+    0 => array:3 [
+      "host" => "127.0.0.1"
+      "port" => 11211
+      "type" => "TCP"
+    ]
+    1 => array:3 [
+      "host" => "127.0.0.2"
+      "port" => 11212
+      "type" => "TCP"
+    ]
+  ]
+  options: {}
+}
+EOTXT;
+        $this->assertDumpEquals($expected, $var);
+    }
+
+    public function testCastMemcachedWithCustomOptions()
+    {
+        if (!class_exists('Memcached')) {
+            $this->markTestSkipped('Memcached not available');
+        }
+
+        $var = new \Memcached();
+        $var->addServer('127.0.0.1', 11211);
+        $var->addServer('127.0.0.2', 11212);
+
+        $var->setOptions($this->getCustomOptions());
+
+        $expectedOptions = $this->getExpectedOptions($var);
+
+        $expectedOptionsAsString = implode(PHP_EOL, $expectedOptions);
+
+        $expected = <<<EOTXT
+Memcached {
+  servers: array:2 [
+    0 => array:3 [
+      "host" => "127.0.0.1"
+      "port" => 11211
+      "type" => "TCP"
+    ]
+    1 => array:3 [
+      "host" => "127.0.0.2"
+      "port" => 11212
+      "type" => "TCP"
+    ]
+  ]
+  options: {
+$expectedOptionsAsString
```

Comment:
I feel like the logic involved into computing this value runs in circles: it expects that its output matches its output - this will always be true even if we introduce a bug...
What about using `assertDumpMatchesFormat` instead and hardcode the expectation using placeholders where things can vary? (could be similar in other test cases you created in previous PRs btw)